flocker/control/_model.py

```diff
@@ -17,27 +17,72 @@
 from characteristic import attributes
 
 from twisted.python.filepath import FilePath
+
 from pyrsistent import (
     pmap, PRecord, field, PMap, CheckedPSet, CheckedPMap,
+    optional as optional_type
     )
 
 from zope.interface import Interface, implementer
 
 
-def pset_field(klass):
+def pset_field(item_type, optional=False):
```

Comment:
`optional` is what pyrsistent calls it. The pyrsistent author has already stated he's willing to accept this upstream, so I decided to stick to what would make sense there.